modules/standard/IO.chpl

```diff
@@ -5428,6 +5428,72 @@ private proc openReaderHelper(path:string,
                              deserializer=deserializer);
 }
 
+/*
+  Create a :record:`fileReader` around a :type:`string`
+
+  Note that the string is copied into a local memory file, so it can be modified
+  after the ``fileReader`` is created without affecting the contents of the
+  ``fileReader``.
+
+  :arg s: the ``string`` to read from
+  :arg deserializer: deserializer to use when reading.
+
+  :returns: a ``fileReader`` reading from the string
+
+*/
+@unstable("'openStringReader' is an experimental feature; its name and behavior are subject to change")
+proc openStringReader(in s: string, in deserializer: ?dt = defaultSerializeVal(false)): fileReader(false, dt) throws {
+  // populate a memory file with the contents of the string
+  const slocal = s.localize();
+  var f = openMemFile(),
+      w = f.writer();
+  w.write(slocal);
+  w.close();
```

Comment:
@jeremiah-corrado I would expect `w.write(s)` to localize automatically as needed. For example, this works, or is it just me?
```chpl
use IO;
var f = open("vass", ioMode.cw);
var w = f.writer();
on Locales[2] {
  var s = "my string\n";
  on Locales[0] {
    w.write(s);
  }
}
```

Also I expect that a `const` intent is more appropriate here, whether you localize explicitly or not. This function merely reads the string, so there is no need to create a copy of the actual argument [when the actual is not dead].